pkg/virt-controller/watch/vmi_test.go

```diff
@@ -1282,6 +1283,10 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			if vmExists {
 				vmSource.Add(vm)
+				// the controller isn't using informer callbacks for the VM informer
```

Comment:
the VM controller and VMI controller share the same VM informer. There has to be a VM in the informer cache for the VM controller to create the VMI.

I do need to add a `vmInfomer.HasSynced()` at the vmi controller startup though to prevent the VMI controller from starting until the VM informer is up to date.

Reply:
I see your point, so basically this can only happen at the test since the VM and VMI are fed to the informers "in parallel".
Anyway, looking at the code, you've already added a `vmInformer.HasSynced()` to the VMI controller's `Run()` method, so it's good to go.

Reply:
> I do need to add a vmInfomer.HasSynced() at the vmi controller startup though to prevent the VMI controller from starting until the VM informer is up to date.

ah, i already did that, never mind